pkg/kubelet/kubelet.go

```diff
@@ -2210,10 +2271,36 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 			// pods that are alive.
 			activePods := kl.filterOutInactivePods(existingPods)
 
-			// Check if we can admit the pod; if not, reject it.
-			if ok, reason, message := kl.canAdmitPod(activePods, pod); !ok {
-				kl.rejectPod(pod, reason, message)
-				continue
+			if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+				// To handle kubelet restarts, test pod admissibility using ResourcesAllocated values
+				// (for cpu & memory) from checkpoint store. If found, that is the source of truth.
+				checkpointState := kl.statusManager.State()
```

Comment:
no action yet:
just leaving a note for myself to track how this state is handled and what happens when missing.